vespaclient-core/src/main/java/com/yahoo/feedhandler/VespaFeedHandler.java

```diff
@@ -30,68 +25,50 @@
+    public static VespaFeedHandler createFromContext(FeedContext context) {
+        return new VespaFeedHandler(context);
     }
 
-    public HttpResponse handle(HttpRequest request, RouteMetricSet.ProgressCallback callback, int numThreads) {
-        try {
-            int busy = busyThreads.incrementAndGet();
-            if (busy > maxBusyThreads)
-                return new EmptyResponse(com.yahoo.jdisc.http.HttpResponse.Status.SERVICE_UNAVAILABLE);
-
-            MessagePropertyProcessor.PropertySetter properties = getPropertyProcessor().buildPropertySetter(request);
+    public FeedResponse handle(HttpRequest request, RouteMetricSet.ProgressCallback callback, int numThreads) {
+        MessagePropertyProcessor.PropertySetter properties = getPropertyProcessor().buildPropertySetter(request);
 
-            String route = properties.getRoute().toString();
-            FeedResponse response = new FeedResponse(new RouteMetricSet(route, callback));
+        String route = properties.getRoute().toString();
+        FeedResponse response = new FeedResponse(new RouteMetricSet(route, callback));
 
-            SingleSender sender = new SingleSender(response, getSharedSender(route));
-            sender.addMessageProcessor(properties);
-            sender.addMessageProcessor(new DocprocMessageProcessor(getDocprocChain(request), getDocprocServiceRegistry(request)));
-            ThreadedFeedAccess feedAccess = new ThreadedFeedAccess(numThreads, sender);
-            Feeder feeder = createFeeder(feedAccess, request);
-            feeder.setAbortOnDocumentError(properties.getAbortOnDocumentError());
-            feeder.setCreateIfNonExistent(properties.getCreateIfNonExistent());
-            response.setAbortOnFeedError(properties.getAbortOnFeedError());
+        SingleSender sender = new SingleSender(response, getSharedSender(route));
+        sender.addMessageProcessor(properties);
+        ThreadedFeedAccess feedAccess = new ThreadedFeedAccess(numThreads, sender);
+        Feeder feeder = createFeeder(feedAccess, request);
+        feeder.setAbortOnDocumentError(properties.getAbortOnDocumentError());
+        feeder.setCreateIfNonExistent(properties.getCreateIfNonExistent());
+        response.setAbortOnFeedError(properties.getAbortOnFeedError());
 
-            List<String> errors = feeder.parse();
-            for (String s : errors) {
-                response.addXMLParseError(s);
-            }
-            if (errors.size() > 0 && feeder instanceof XMLFeeder) {
-                response.addXMLParseError("If you are trying to feed JSON, set the Content-Type header to application/json.");
-            }
+        List<String> errors = feeder.parse();
+        for (String s : errors) {
+            response.addXMLParseError(s);
+        }
+        if (errors.size() > 0 && feeder instanceof XMLFeeder) {
+            response.addXMLParseError("If you are trying to feed JSON, set the Content-Type header to application/json.");
```

Comment:
Consider updating error message to indicate appropriate command line option instead of referring to HTTP headers (or just remove it entirely)

Reply:
Removed :)